BrainPortal/spec/boutiques/boutiques_tester_spec.rb

```diff
@@ -47,9 +47,64 @@
 # Run the Boutiques test on the BrainPortal side
 describe "BrainPortal Boutiques Tests" do
 
-  # Run before block to create required input files
   before(:all) do
-    createInputFiles
+    createInputFiles # Create required input files
+    PWD = Dir.pwd # Save the starting dir form which the tests were launched
+  end
+
+  before(:each) do
+    # Change to the temporary storage directory so files are automatically written there
+    Dir.chdir TempStore
+    # Build some of the cbrain environment
+    @user, @group = FactoryGirl.create(:user), FactoryGirl.create(:group)
+    @dp = FlatDirLocalDataProvider.new({
+      :online => true, :read_only => false, :remote_dir => '.', :name => "dp1", :user_id => @user.id, :group_id => @group.id
+    })
+    @dp.save!
+    # Lambda for constructing cbcsv files
+    @makeCbcsv = -> (fs, name, task, mangler=nil, addToIuids=true, user: @user, group: @group, dp: @dp) {
+      flist = CbrainFileList.new(
+        :user             => user,
+        :user_id          => user.id,
+        :group_id         => group.id,
+        :name             => name,
+        :data_provider_id => dp.id,
+      )
+      flist.save
+      text = CbrainFileList.create_csv_file_from_userfiles( fs )
+      text = mangler.( text ) unless mangler.nil?
+      flist.cache_writehandle { |t| t.write( text ) }
+      task.params[:interface_userfile_ids] << flist.id if addToIuids
+      flist # return the cbcsv object
+    }
+    # Helper for generating and sending userfiles to tasks
+    @addUserFile = -> (name, task, addToUids=true, user: @user, group: @group, dp: @dp) {
+      uf = SingleFile.new({data_provider_id: dp.id, name: name, group_id: group.id, user_id: user.id})
+      uf.save!
+      task.params[:interface_userfile_ids] << uf.id if addToUids
+      uf
+    }
+    # Checks after_form output. Checks both the number or errors and ascertains at least one of their contents.
+    @checkAfterForm = -> (task, checkVal=0, atLeastOneErrWith=nil, runBeforeForm=false) {
+      task.before_form if runBeforeForm
+      task.after_form # Run the method
+      errMsgs = task.params_errors.full_messages # Get any error messages
+      expect( errMsgs.any? { |e| e.include? atLeastOneErrWith } ).to be true unless atLeastOneErrWith.nil?
+      expect( errMsgs.length ).to eq( checkVal )
+    }
+    # Function to manually add a null row to a cbcsv
+    @nilRowAdder = -> (text) {
+      fs, rt, n = CbrainFileList::FIELD_SEPARATOR, CbrainFileList::RECORD_TERMINATOR, CbrainFileList::ATTRIBUTES_LIST.length
+      splitTxt  = text.split( rt )
+      nilRow    = '0' + fs*(n - 1)
+      splitTxt  << nilRow
+      splitTxt.join( rt )
```

Comment:
You need a record terminator after the last record too... `join(rt)` will not do that.

You could simply do

```ruby
splitTxt << ""
````

just before the line with `join()` though.